dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -2249,6 +2416,1221 @@ async fn cmd_db_network_list_vnics(
+        })
+        .await?;
+
+    check_limit(&sagas, fetch_opts.fetch_limit, || {
+        String::from("listing stuck sagas")
+    });
+
+    print_sagas(sagas, false);
+
+    Ok(())
+}
+
+async fn get_saga(
+    datastore: &DataStore,
+    saga_id: Uuid,
+) -> Result<Saga, anyhow::Error> {
+    use db::schema::saga::dsl;
+
+    let saga = dsl::saga
+        .filter(dsl::id.eq(saga_id))
+        .first_async(&*datastore.pool_connection_for_tests().await?)
+        .await?;
+
+    Ok(saga)
+}
+
+/// Show the execution of a saga
+async fn cmd_db_sagas_show(
+    datastore: &DataStore,
+    fetch_opts: &DbFetchOptions,
+    args: &SagaShowArgs,
+) -> Result<(), anyhow::Error> {
+    use db::schema::saga_node_event::dsl;
+
+    let saga_nodes = db::paginated(
+        dsl::saga_node_event,
+        dsl::event_time,
+        &first_page::<dsl::event_time>(fetch_opts.fetch_limit),
+    )
+    .filter(dsl::saga_id.eq(args.saga_id))
+    .order_by(dsl::event_time)
+    .load_async(&*datastore.pool_connection_for_tests().await?)
+    .await?;
+
+    print_saga_nodes(
+        Some(get_saga(datastore, args.saga_id).await?),
+        saga_nodes,
+    );
+
+    Ok(())
+}
+
+/// List all sagas that unwound sucessfully
+async fn cmd_db_sagas_list_unwound(
+    datastore: &DataStore,
+    fetch_opts: &DbFetchOptions,
+) -> Result<(), anyhow::Error> {
+    let sagas = datastore
+        .pool_connection_for_tests()
+        .await?
+        .transaction_async(|conn| async move {
+            use db::schema::saga_node_event::dsl;
+
+            // Sorting by time_created requires a full table scan with the
+            // current index definitions.
+            conn.batch_execute_async(ALLOW_FULL_TABLE_SCAN_SQL).await?;
+
+            let undo_failed_sagas: HashSet<SagaId> = db::paginated(
+                dsl::saga_node_event,
+                dsl::event_time,
+                &first_page::<dsl::event_time>(fetch_opts.fetch_limit),
+            )
+            .filter(dsl::event_type.eq(String::from("undo_failed")))
+            .select(dsl::saga_id)
+            .order_by(dsl::event_time)
+            .load_async(&conn)
+            .await?
+            .into_iter()
+            .collect();
+
+            let undo_started_sagas: HashSet<SagaId> = db::paginated(
+                dsl::saga_node_event,
+                dsl::event_time,
+                &first_page::<dsl::event_time>(fetch_opts.fetch_limit),
+            )
+            .filter(dsl::event_type.eq(String::from("undo_started")))
+            .select(dsl::saga_id)
+            .order_by(dsl::event_time)
+            .load_async(&conn)
+            .await?
+            .into_iter()
+            .collect();
+
+            // A saga successfully unwound if it started unwinding and didn't
+            // fail for any of the nodes.
+            let sagas: Vec<_> = undo_started_sagas
+                .difference(&undo_failed_sagas)
+                .cloned()
+                .collect();
```

Comment:
This seems like an example of steno internals that could change.  Just thinking out loud: what if instead of either duplicating or exporting Steno-internal types, it had a way of loading up records for a saga (i.e., you'd give it the same objects that it gives you in https://docs.rs/steno/latest/steno/trait.SecStore.html) and then you could ask it the various questions that you're trying to ask here (like: did this one unwind? did it get stuck? what's the name of this node?)